pkg/server/downloader.go

```diff
@@ -16,157 +16,156 @@ import (
 	"github.com/longhorn/backing-image-manager/pkg/util"
 )
 
-type DownloaderFactory interface {
-	NewDownloader() Downloader
-}
-
-type Downloader interface {
-	GetSize(url string) (size int64, err error)
-	Pull(url, filepath string, updater util.ProgressUpdater) (written int64, err error)
-	Receive(port string, filePath string, syncFileOps sparserest.SyncFileOperations) error
-	Send(filePath string, address string) error
-	Cancel()
-}
-
-type BackingImageDownloaderFactory struct{}
-
 type BackingImageDownloader struct {
 	*sync.RWMutex
-	cancelFunc context.CancelFunc
+	isDownloading bool
+	ctx           context.Context
+	cancelFunc    context.CancelFunc
+
+	// for unit test
+	getSizeFunc        func(url string) (int64, error)
```

Comment:
IMO, for this kind of function attributes/properties, it's better to organize in an interface. But, it's OK for now.

Reply:
Will abstract these functions to an interface.